packages/apputils-extension/src/shortcuts.tsx

```diff
@@ -0,0 +1,148 @@
+
+export function displayShortcuts(
+  app: JupyterFrontEnd,
+  trans: TranslationBundle
+) {
+  const { commands } = app;
+  const elt = document.activeElement;
+
+  /**
+   * Find the distance from the target node to the first matching node.
+   *
+   * Based on Lumino private function commands.Private.targetDistance
+   * This traverses the DOM path from `elt` to the root
+   * computes the distance from `elt` to the first node which matches
+   * the CSS selector. If no match is found, `-1` is returned.
+   *
+   * It also stops traversal if the `data-lm-suppress-shortcuts` or
+   * `data-p-suppress-shortcuts` attributes are found.
+   */
+  function matchDistance(selector: string, elt: Element | null): number {
+    let targ = elt;
+    for (
+      let dist = 0;
+      targ !== null && targ !== targ.parentElement;
+      targ = targ.parentElement, ++dist
+    ) {
+      if (targ.hasAttribute('data-lm-suppress-shortcuts')) {
+        return -1;
+      }
+      /* <DEPRECATED> */
+      if (targ.hasAttribute('data-p-suppress-shortcuts')) {
+        return -1;
+      }
+      /* </DEPRECATED> */
+      if (targ.matches(selector)) {
+        return dist;
+      }
+    }
+    return -1;
+  }
+
+  function formatKeybinding(keys: readonly string[]): string {
+    return keys.map(CommandRegistry.formatKeystroke).join(', ');
+  }
+
+  // Find active keybindings for target element
+  const activeBindings = new Map<
+    string,
+    [number, CommandRegistry.IKeyBinding]
+  >();
+  for (let i = 0; i < commands.keyBindings.length; i++) {
+    const kb = commands.keyBindings[i];
+    let distance = matchDistance(kb.selector, elt);
+    if (distance < 0) {
+      continue;
+    }
+    let formatted = formatKeybinding(kb.keys);
+    if (activeBindings.has(formatted)) {
+      let oldBinding = activeBindings.get(formatted)!;
+      // if the existing binding takes precedence, ignore this binding by continuing
+      if (
+        oldBinding[0] < distance ||
+        (oldBinding[0] === distance &&
+          Selector.calculateSpecificity(oldBinding[1].selector) >
+            Selector.calculateSpecificity(kb.selector))
+      ) {
+        continue;
+      }
+    }
+    activeBindings.set(formatted, [distance, kb]);
+  }
+
+  // Group shortcuts by distance
+  let maxDistance = -1;
+  const groupedBindings = new Map<number, CommandRegistry.IKeyBinding[]>();
+  for (let [distance, binding] of activeBindings.values()) {
+    maxDistance = distance > maxDistance ? distance : maxDistance;
+    if (!groupedBindings.has(distance)) {
+      groupedBindings.set(distance, []);
+    }
+    groupedBindings.get(distance)!.push(binding);
+  }
+
+  // Display shortcuts by group
+  const bindingTable = [];
+  for (let d = 0; d <= maxDistance; d++) {
+    if (groupedBindings.has(d)) {
+      console.log(`Binding level ${d}:`);
+      bindingTable.push(
+        groupedBindings.get(d)!.map(b => (
+          <tr key={b.command}>
+            <td
+              style={{
```

Comment:
We should prefer defining styles in CSS files to enable consistent theming and use our CSS variables whenever possible (font size, colour, etc).

Reply:
@krassowski Thanks for the comments. They should be taken into account in 
a72ed0c.